csi.proto

```diff
@@ -248,6 +248,22 @@ message VolumeInfo {
   // subsequent calls to refer to the provisioned volume. This field
   // should not exceed 1MiB.
   string id = 2;
+
+  // Attributes reflect static properties of a volume and MUST be passed
+  // to volume validation and publishing calls. Attributes MAY NOT
+  // uniquely identify a volume. A volume uniquely identified by `id`
+  // SHALL always report the same attributes. This field is REQUIRED.
```

Comment:
During the meeting, I think we said that `attributes` was required to be passed by the CO iff it is returned by the SP.  We seem to be losing that nuance in this spec and declaring it as required always.  Does that mean that we will always require the SP to return a map, and in the case where the SP doesn't care about parameters, it should just return an empty map?  I guess I am OK with that requirement, but just want to make sure we are all on the same page.

Reply:
Hi @julian-hj,

If it's on the SP to indicate if the data is required it can return an error if an empty map is transmitted, right? If you're just wanting to clarify the spec then I agree it needs to have tighter language.

Reply:
I think in @jdef's PR, it's a struct `Attributes`. So it's required to set this field, but the map inside the struct can be empty. If we change this to `map<string, string>`, we need to clarify in the wording. This field is indeed optional, but if set, the CO should pass that to the subsequent calls.

Reply:
@jieyu Ah, that makes sense.  So we didn't lose the nuance, we just codified it as an optional map inside a required struct.  Makes sense.  I still think we'd be better off with a plain map & some explicit language about the behavior though.